dashboard/test/lib/pd/survey_pipeline/survey_rollup_decorator_test.rb

```diff
@@ -0,0 +1,114 @@
+require 'test_helper'
+require 'pd/survey_pipeline/survey_rollup_decorator.rb'
+
+module Pd::SurveyPipeline
+  class SurveyRollupDecoratorTest < ActiveSupport::TestCase
+    test 'decorate empty survey roll-up results' do
+      facilitator = create :facilitator
+      workshop_id = 1
+      related_workshop_ids = [workshop_id]
+
+      # There is no survey submission for this combination of workshop and facilitator
+      input_data = {
+        current_workshop_id: workshop_id,
+        related_workshop_ids: related_workshop_ids,
+        facilitator_id: facilitator.id,
+        parsed_questions: {},
+        question_categories: %w(overall_success teacher_engagement facilitator_effectiveness),
+        workshop_submissions: [],
+        facilitator_submissions: [],
+        question_answer_joined: [],
+        summaries: [],
+        errors: []
+      }
+
+      expected_result = {
+        facilitators: {facilitator.id => facilitator.name},
+        current_workshop: workshop_id,
+        related_workshops: {facilitator.id => related_workshop_ids},
+        facilitator_averages: {
+          questions: {},
+          facilitator.name => {
+            'overall_success' => {this_workshop: nil, all_my_workshops: nil},
+            'teacher_engagement' => {this_workshop: nil, all_my_workshops: nil},
+            'facilitator_effectiveness' => {this_workshop: nil, all_my_workshops: nil}
+          }
+        },
+        facilitator_response_counts: {
+          this_workshop: {facilitator.id => {}},
+          all_my_workshops: {facilitator.id => {}}
+        },
+        errors: []
+      }
+
+      result = SurveyRollupDecorator.decorate_facilitator_rollup input_data
+
+      assert_equal expected_result, result
+    end
+
+    test 'decorate simple non-empty survey roll-up results' do
+      facilitator = create :facilitator
+      workshop = create :csp_academic_year_workshop
+      # Both workshops are of the same kind and (implicitly) have the same facilitator
```

Comment:
if it is necessary for this test that the workshops have the same facilitator, let's be explicit and set that up here, in case factory behavior changes.

Reply:
I will make it explicit